amf-aml/shared/src/main/scala/amf/aml/internal/parse/instances/parser/ObjectPropertyParser.scala

```diff
@@ -11,7 +11,7 @@ object ObjectPropertyParser extends NodeMappableHelper {
 
   def parse[T <: DomainElement](id: String,
                                 propertyEntry: YMapEntry,
-                                property: PropertyMapping,
+                                property: PropertyLikeMapping[_],
```

Comment:
Does this generalization make sense? This should always parse PropertyMappings

Reply:
Yes this is a weird change. Had to do this because when parsing a semantic extension I have an annotation mapping and not a property mapping, therefore I had to change the signatures of all parsing methods.
The difference between an AnnotationMapping and a PropertyMapping is very thin (think the only difference is that the PopertyMapping can have mapKey and mapValue fields and the annotation mapping has the domain to which it applies)

For me the change is ok.